examples/bri-3/src/bri/identity/bpiSubjectAccounts/models/bpiSubjectAccount.ts

```diff
@@ -1,14 +1,12 @@
-import { IBpiSubject } from "../../bpiSubjects/models/bpiSubject.interface";
-import { IBpiSubjectAccount } from "./bpiSubjectAccount.interface";
+import { BpiSubject } from '../../bpiSubjects/models/bpiSubject';
 
-export class BpiSubjectAccount implements IBpiSubjectAccount {
+export class BpiSubjectAccount {
     id: string; // TODO: Add uuid after #491
-    creatorBpiSubject: IBpiSubject;
-    ownerBpiSubject: IBpiSubject;
+    creatorBpiSubject: BpiSubject;
+    ownerBpiSubject: BpiSubject;
     bpiAccounts: string[]; // TODO: Add as part of #489
 
-    constructor(creatorBpiSubject: IBpiSubject, ownerBpiSubject: IBpiSubject) {
-        this.creatorBpiSubject = creatorBpiSubject;
-        this.ownerBpiSubject = ownerBpiSubject;
+    constructor(init: BpiSubjectAccount) {
+        Object.assign(this, init);
```

Comment:
is this safe to use object.assign like this? it is not type safe, and init object can also pass in arbitrary id like this, and maybe some other side effects

Reply:
yeah this is one of the new implementations from Typescript 2.1
The init object is the class type, so this is type safe.

Reply:
How is init variable instantiated? We need a ctor that accepts the basic fields as params. We can later on include a ctor that instantiates another instance from a entity model, once we introduce those.

Reply:
in order to instantiate the partial typed objects for class A, the calling will be:

let obj : A = return new A(<A>{...})